lib/parsers/mpeg_parser.rb

```diff
@@ -0,0 +1,115 @@
+
+# MPEG Headers documentation:
+# http://dvd.sourceforge.net/dvdinfo/mpeghdrs.html#seq
+# http://www.cs.columbia.edu/~delbert/docs/Dueck%20--%20MPEG-2%20Video%20Transcoding.pdf
+# Useful tool to check the file information: https://www.metadata2go.com/
+class FormatParser::MPEGParser
+  include FormatParser::IOUtils
+
+  MPEG_VALUES = {
+    'mpg' => :mpg,
+    'mpeg' => :mpeg
+  }
+
+  ASPECT_RATIOS = {
+    1 => '1:1',
+    2 => '4:3',
+    3 => '16:9',
+    4 => '2.21:1'
+  }
+
+  FRAME_RATES = {
+    1 => '23.976',
+    2 => '24',
+    3 => '25',
+    4 => '29.97',
+    5 => '30',
+    6 => '50',
+    7 => '59.94',
+    8 => '60'
+  }
+
+  PACK_HEADER_START_CODE = '000001ba'.freeze
```

Comment:
You can directly use strings in binary encoding to signify bytes, and compare directly on them. This will free you up from the need to do `to_hex` and back on every comparison / print, but it will also indicate to the reader how many bytes you are planning to compare. If you need a sequence of bytes and it is easy to define it in hexadecimal you can use this:

```
[0x00, 0x00, 0x01, 0xBA].pack("C*")
```

Reply:
Thanks! That works :)